pkg/search/postgres/common.go

```diff
@@ -1179,13 +1179,21 @@ func retryableGetCursorSession(ctx context.Context, schema *walker.Schema, q *v1
 
 	queryStr := preparedQuery.AsSQL()
 
-	tx, err := db.Begin(ctx)
-	if err != nil {
-		return nil, errors.Wrap(err, "creating transaction")
+	tx, ok := postgres.TxFromContext(ctx)
```

Comment:
nit: if we are going to use `ok` elsewhere we should probably name it something meaningful like `parentTxExists` or something.